vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go

```diff
@@ -220,35 +263,54 @@ func getPodID(pod *apiv1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
-func getReplicaCount(creator podReplicaCreator, client kube_client.Interface) (int, error) {
+func (f *podsEvictionRestrictionFactoryImpl) getReplicaCount(creator podReplicaCreator) (int, error) {
 	switch creator.Kind {
-	case "ReplicationController":
-		rc, err := client.CoreV1().ReplicationControllers(creator.Namespace).Get(creator.Name, metav1.GetOptions{})
-
-		if err != nil || rc == nil {
+	case replicationController:
+		rcObj, exists, err := f.rcInformer.GetStore().GetByKey(creator.Namespace + "/" + creator.Name)
+		if err != nil || rcObj == nil {
```

Comment:
Shouldn't the == nil check be done together with !exists.

Reply:
I think this check is redundant anyway. Removed.